src/ck-ldb/ckgraph.h

```diff
@@ -14,159 +14,164 @@
 #ifndef _CKGRAPH_H_
 #define _CKGRAPH_H_
 
-#include <vector>
 #include "BaseLB.h"
+#include <vector>
 
-class ProcInfo {
+class ProcInfo
+{
   friend class ProcArray;
 
-  public:
-    ProcInfo() {}
-    ProcInfo(int i, double ov, double tl, double sp, bool avail): id(i), _overhead(ov), _totalLoad(tl), _pe_speed(sp), available(avail) {}
-    inline int getProcId() { return id; }
-    inline void setProcId(int _id) { id = _id; }
-    inline double getTotalLoad() const { return _totalLoad; }
-//    inline void setTotalLoad(double load) { totalLoad = load; }
-//    inline double getOverhead() { return overhead; }
-//    inline void setOverhead(double oh) { overhead = oh; }
-    inline double &overhead() { return _overhead; }
-    inline double &totalLoad() { return _totalLoad; }
-    inline double &pe_speed() { return _pe_speed; }
-    inline bool isAvailable() { return available; }
-
-  protected:
-    int id;		// CkMyPe of the processor
-    double _overhead;	// previously called background load (bg_walltime)
-    double _totalLoad;	// includes object_load + overhead
-    double _pe_speed;	// CPU speed
-    bool available;	// if the processor is available
+public:
+  ProcInfo() = default;
+  ProcInfo(int i, double ov, double tl, double sp, bool avail)
+      : id(i), _overhead(ov), _totalLoad(tl), _pe_speed(sp), available(avail)
+  {
+  }
+  int getProcId() const { return id; }
+  void setProcId(int _id) { id = _id; }
+  double getTotalLoad() const { return _totalLoad; }
```

Comment:
Why two accessors for total load?

Reply:
Also as a followup...do we even want the accessors by reference? are they necessary compared to just get/set methods like are used everywhere else here?

Reply:
Agreed that the accessors need a little bit more cleanup. At the very least, if we want to keep the ones returning by reference, there should be const-qualified versions that return const references. But I'm not sure if we need accessors returning by reference at all.

Reply:
Good call, I've removed the ones that return by reference, I don't think they're needed at all.